lib/connectHapiServerToRedis.js

```diff
@@ -42,17 +44,38 @@ function connectHapiServerToRedis (server, cb) {
       authenticate: function(req, cb) {
 		  
         // Get cookie id
-		
-        // Call redis with cookie to get session
-		
-        // From session get userid
+    
         var rawSid = req.state['connect.sid.1.0'];
+
+        if (!rawSid) {
+          console.log('Cookie does not exist!');
+          return cb( new Boom.unauthorized('Not authorized'));
+        }
+
+        // Get session ID and verify signature
  
-        //var sid = RedisSessionReader.unsignCookieSid(rawSid, secret);
-        //RedisSessionReader.getSessionIdFromRedis(sessionId, server.plugins['hapi-redis'].client, function (err, userId) {
-        //  if (err) {return cb(err);}
-        //  cb(null, { userId: userId })
-        //});		  
+        var sid = unsignCookieSid(rawSid, secret);
+
+        if (!sid) {
+
+          console.log('Cookie not signed so not authorized');
+          return cb( new Boom.unauthorized('Not authorized'));
+
+        }
```

Comment:
I guess this makes sense.  This is definitely convenient -- the only disadvantageous I can think of is that we're locking ourselves into returning a 401 anytime the user isn't authenticated. What if at some point in the future, there's a request that treats a user who is logged in differently than a user who isn't logged in (for example, maybe the logged in user gets more data than the one who isn't)?  But maybe that's something we shouldn't worry about right now...